lib/dal/tests/integration_test/property_editor.rs

```diff
@@ -32,3 +33,261 @@ async fn assemble(ctx: &DalContext) {
         .expect("could not assemble property editor schema");
     dbg!(property_editor_schema, property_editor_values);
 }
+
+#[test]
+async fn array_map_manipulation(ctx: &DalContext) {
+    // Create a component using the testing schema
+    let component = create_component_for_schema_name(ctx, "pirate", "ss poopcanoe").await;
+    // let variant_id = Component::schema_variant_id(ctx, component.id())
+    //     .await
+    //     .expect("find variant id for component");
+
+    let parrot_names_value_id = component
+        .attribute_values_for_prop(ctx, &["root", "domain", "parrot_names"])
+        .await
+        .expect("find value ids for the prop parrot_names")
+        .pop()
+        .expect("there should only be one value id");
+
+    let treasure_map_value_id = component
+        .attribute_values_for_prop(ctx, &["root", "domain", "treasure"])
+        .await
+        .expect("find value ids for the prop treasure")
+        .pop()
+        .expect("there should only be one value id");
+
+    // Add items to array prop
+    AttributeValue::insert(ctx, parrot_names_value_id, Some("tabitha".into()), None)
+        .await
+        .expect("one item in array");
+    AttributeValue::insert(ctx, parrot_names_value_id, Some("samantha".into()), None)
+        .await
+        .expect("two items in array");
+    AttributeValue::insert(ctx, parrot_names_value_id, Some("jessica".into()), None)
+        .await
+        .expect("three items in array");
+    AttributeValue::insert(ctx, parrot_names_value_id, Some("amanda".into()), None)
+        .await
+        .expect("four items in array");
+    AttributeValue::insert(ctx, parrot_names_value_id, Some("dr wiggles".into()), None)
+        .await
+        .expect("five items in array");
+
+    // Add items to map prop
+    AttributeValue::insert(
+        ctx,
+        treasure_map_value_id,
+        Some("cheese".into()),
+        Some("ohio".to_string()),
+    )
+    .await
+    .expect("one item in map");
+    AttributeValue::insert(
+        ctx,
+        treasure_map_value_id,
+        Some("coxinha".into()),
+        Some("rio".to_string()),
+    )
+    .await
+    .expect("two items in map");
+    AttributeValue::insert(
+        ctx,
+        treasure_map_value_id,
+        Some("pizza".into()),
+        Some("nyc".to_string()),
+    )
+    .await
+    .expect("three items in map");
+    AttributeValue::insert(
+        ctx,
+        treasure_map_value_id,
+        Some("sushi".into()),
+        Some("tokyo".to_string()),
+    )
+    .await
+    .expect("four items in map");
+    AttributeValue::insert(
+        ctx,
+        treasure_map_value_id,
+        Some("baby back ribs".into()),
+        Some("jupiter".to_string()),
+    )
+    .await
+    .expect("five items in map");
+
+    // Grab the children for the array and check that they match what they should be
```

Comment:
Normally, we would want to call `ctx.blocking_commit()` before performing assertions on anything we have built. That way, we can let dependent values update run and the rebaser work.

_That being said_, since this test only inserts and removes attribute values and doesn't update them, I believe this should be fine.